plugins/obs-filters/color-grade-filter.c

```diff
@@ -292,16 +286,46 @@ static void color_grade_filter_update(void *data, obs_data_t *settings)
 			filter->target = make_clut_texture_png(
 				filter->image.format, filter->image.cx,
 				filter->image.cy, filter->image.texture_data);
-			filter->clut_scale =
-				(float)(LUT_WIDTH - 1) / (float)LUT_WIDTH;
-			filter->clut_offset = 0.5f / (float)LUT_WIDTH;
+			const float clut_scale = (float)(LUT_WIDTH - 1);
+			vec3_set(&filter->clut_scale, clut_scale, clut_scale,
+				 clut_scale);
+			vec3_set(&filter->clut_offset, 0.f, 0.f, 0.f);
 		} else if (filter->cube_data) {
 			const uint32_t width = filter->cube_width;
-			filter->target = gs_voltexture_create(
-				width, width, width, GS_RGBA16F, 1,
-				(uint8_t **)&filter->cube_data, 0);
-			filter->clut_scale = (float)(width - 1) / (float)width;
-			filter->clut_offset = 0.5f / (float)width;
+			if (filter->clut_dim == CLUT_1D) {
+				filter->target = gs_texture_create(
+					width, 1, GS_RGBA16F, 1,
+					(uint8_t **)&filter->cube_data, 0);
```

Comment:
Use correct const-ness to resolve compiler warnings.
(const uint8_t **)&filter->cube_data

Reply:
Having to do this extra const cast seems like a silly oversight in the C spec that they should just copy over from C++. Added.

Reply:
I agree, and im sure they'll fix it in C20 :)

Reply:
Yeah, but MS won't implement it so you'll be stuck doing this on Windows forever. ;-)